server/service/src/invoice/outbound_shipment/update/generate.rs

```diff
@@ -77,7 +81,7 @@ fn should_update_batches(
         let new_invoice_status_index = new_invoice_status.index();
 
         new_invoice_status_index >= InvoiceRowStatus::Picked.index()
-            && invoice_status_index < new_invoice_status_index
+            && invoice_status_index < InvoiceRowStatus::Picked.index()
```

Comment:
actual fix. 
btw this method was refactored a few times, wanted to check if it was the refactor that cause the problem, it did seem like it was the refactor that cause the issue, but the refactor was done because more statuses was added, it looked like this at the start:
```
pub fn should_update_batches(invoice: &InvoiceRow, patch: &UpdateOutboundShipment) -> bool {
    match (&invoice.status, &patch.status) {
        (InvoiceRowStatus::Draft, Some(InvoiceStatus::Confirmed)) => true,
        (InvoiceRowStatus::Draft, Some(InvoiceStatus::Finalised)) => true,
        _ => false,
    }
}
```

Might have been safer to use above type of check of status, it would mean two extra match clauses. But I think the safest would have been to add more thorough test cases for this sort of side effect (i.e. write test not just to validate that it works, but to make sure it's not broken in the future)